fdbclient/BlobStore.actor.cpp

```diff
@@ -218,11 +222,15 @@ Reference<BlobStoreEndpoint> BlobStoreEndpoint::fromString(std::string const &ur
 		if(resourceFromURL != nullptr)
 			*resourceFromURL = resource.toString();
 
-		StringRef c(cred);
-		StringRef key = c.eat(":");
-		StringRef secret = c.eat();
+		Optional<BlobStoreEndpoint::Credentials> creds;
+		if (cred.present()) {
+			StringRef c(cred.get());
```

Comment:
The suggestion above removes the need for this line or the additional variable `c`.